Libraries/Components/PasswordBoxWindows/PasswordBoxWindows.windows.js

```diff
@@ -0,0 +1,505 @@
+     * multiline fields. Note that for multiline fields, setting blurOnSubmit
+     * to true means that pressing return will blur the field and trigger the
+     * onSubmitEditing event instead of inserting a newline into the field.
+     */
+    blurOnSubmit: PropTypes.bool,
+    /**
+     * Styles
+     */
+    style: Text.propTypes.style,
+    /**
+     * The color of the textInput underline.
+     * @platform android
+     */
+    underlineColorAndroid: PropTypes.string,
+  },
+
+  /**
+   * `NativeMethodsMixin` will look for this when invoking `setNativeProps`. We
+   * make `this` look like an actual native component class.
+   */
+  mixins: [NativeMethodsMixin, TimerMixin],
+
+  viewConfig: RCTTextBox.viewConfig,
+
+  /**
+   * Returns if the input is currently focused.
+   */
+  isFocused: function(): boolean {
+    return PasswordBoxInputState.currentlyFocusedField() ===
+      ReactNative.findNodeHandle(this.refs.input);
+  },
+
+  contextTypes: {
+    onFocusRequested: React.PropTypes.func,
+    focusEmitter: React.PropTypes.instanceOf(EventEmitter),
+  },
+
+  _focusSubscription: (undefined: ?Function),
+  _lastNativeText: (undefined: ?string),
+
+  componentDidMount: function() {
+    this._lastNativeText = this.props.value;
+    if (!this.context.focusEmitter) {
+      if (this.props.autoFocus) {
+        this.requestAnimationFrame(this.focus);
+      }
+      return;
+    }
+    this._focusSubscription = this.context.focusEmitter.addListener(
+      'focus',
+      (el) => {
+        if (this === el) {
+          this.requestAnimationFrame(this.focus);
+        } else if (this.isFocused()) {
+          this.blur();
+        }
+      }
+    );
+    if (this.props.autoFocus) {
+      this.context.onFocusRequested(this);
+    }
+  },
+
+  componentWillUnmount: function() {
+    this._focusSubscription && this._focusSubscription.remove();
+    if (this.isFocused()) {
+      this.blur();
+    }
+  },
+
+  getChildContext: function(): Object {
+    return {isInAParentText: true};
+  },
+
+  childContextTypes: {
+    isInAParentText: React.PropTypes.bool
+  },
+
+  /**
+   * Removes all text from the input.
+   */
+  clear: function() {
+    this.setNativeProps({text: ''});
+  },
+
+  _getText: function(): ?string {
+    return typeof this.props.value === 'string' ?
+      this.props.value :
+      this.props.defaultValue;
+  },
+
+  render: function() {
+    var textContainer;
+
+    var onSelectionChange;
+    if (this.props.selectionState || this.props.onSelectionChange) {
+      onSelectionChange = (event: Event) => {
+        if (this.props.selectionState) {
+          var selection = event.nativeEvent.selection;
+          this.props.selectionState.update(selection.start, selection.end);
+        }
+        this.props.onSelectionChange && this.props.onSelectionChange(event);
+      };
+    }
+
+    var children = this.props.children;
+    var childCount = 0;
+    ReactChildren.forEach(children, () => ++childCount);
+    invariant(
```

Comment:
>invariant [](start = 4, length = 9)

weird that there are various carry overs of other platforms (ios and android) in the props declarations above, but we have windows-only behavior here.  Since this component only shows up on Windows, I'd try to eliminate anything that is not relevant to windows.